test/checker.go

```diff
@@ -0,0 +1,57 @@
+package test
+
+import (
+	"fmt"
+	"reflect"
+
+	check "gopkg.in/check.v1"
+)
+
+// IsInList Checker
+// The `IsInList` checker verifies that the element is in the container.
+// The container must be a slice or array and the element must be the same type as the container.
+//
+// For example:
+//
+//     c.Assert("b", IsInList, []string{"a", "b", "c"})
+//     c.Assert(2, IsInList, []int{1, 2, 3})
+//
+
+type isInListChecker struct {
+	*check.CheckerInfo
+}
+
+var IsInList check.Checker = &isInListChecker{
+	&check.CheckerInfo{
+		Name:   "IsInList",
+		Params: []string{"element", "container"},
+	},
+}
+
+func (checker *isInListChecker) Check(params []interface{}, names []string) (result bool, error string) {
+	defer func() {
+		if r := recover(); r != nil {
+			result = false
+			error = fmt.Sprint(r)
+		}
+	}()
+
+	element := params[0]
+	container := params[1]
+
+	containerV := reflect.ValueOf(container)
```

Comment:
```suggestion
	valueOfList := reflect.ValueOf(list)
```

Reply:
Updated with the above suggestions! Please help to recheck.